Tools/Scripts/libraries/resultsdbpy/resultsdbpy/view/static/js/dashboard.js

```diff
@@ -348,7 +348,28 @@ class Dashboard {
         if (!this.ref.state.displayed || !this.requestData)
             return;
 
-        const now = Math.floor(Date.now() / 1000);
+        const globalParams = this.documentParams()
+        let now = Math.floor(Date.now() / 1000);
+        if (Object.hasOwn(globalParams, 'before_time'))
+            now = Math.min(now, globalParams.before_time[0]);
+        if (Object.hasOwn(globalParams, 'before_timestamp'))
+            now = Math.min(now, globalPaarams.before_timestamp[0]);
+        if (Object.hasOwn(globalParams, 'before_uuid'))
+            now = Math.min(now, Math.floor(globalParams.before_uuid[0] / 100));
+        if (Object.hasOwn(globalParams, 'before_ref')) {
+            let commit_banck = CommitBank;
+            if (Object.hasOwn(globalParams, 'branch'))
+                this.commit_bank = CommitBank.forBranch(globalParams.branch);
+            const refAsString = `${globalParams.before_ref[0]}`;
+            for (const i in CommitBank.commits) {
+                const commit = CommitBank.commits[i];
```

Comment:
nit: can be simplified to a for-of loop: `for (const commit of CommitBank.commits) { … }`